stdlib/LinearAlgebra/test/blas.jl

```diff
@@ -641,6 +641,30 @@ end
     @test BLAS.get_num_threads() === default
 end
 
+@testset "with_num_threads" begin
+    prev_num_threads = BLAS.get_num_threads()
+    context_num_threads = BLAS.with_num_threads(1) do
+        BLAS.get_num_threads()
+    end
+    @test context_num_threads == 1
+    @test prev_num_threads == BLAS.get_num_threads()
+
+    @testset "thread unsafe" begin
+        prev_num_threads = BLAS.get_num_threads()
+        # thread unsafe
+        @async BLAS.with_num_threads(1) do
```

Comment:
What is this test testing?

Reply:
This is used to simulate the case that when there exists a task with `with_num_threads`, then all threads are affected until that task is finished.